pkg/util/volume.go

```diff
@@ -37,3 +38,46 @@ func VolumeFromName(v volume.VolumeDriver, name string) (*api.Volume, error) {
 	}
 	return nil, fmt.Errorf("Cannot locate volume with name %s", name)
 }
+
+// VolumeFromIdSdk uses the SDK to fetch the volume object associated with the specified id.
+func VolumeFromIdSdk(ctx context.Context, volumes api.OpenStorageVolumeClient, id string) (*api.Volume, error) {
+	inspectResp, err := volumes.Inspect(ctx, &api.SdkVolumeInspectRequest{
+		VolumeId: id,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Cannot locate volume with id %s", id)
```

Comment:
check that these errors or that the caller is compatible with the CSI spec. AFAIK, you can just return the error from the SDK inspect call.

Reply:
Inspect returns NotFound. All `utils.VolumeFromIdSdk`/etc calls return NotFound.

`AlreadyExists` is used if the volume already exists. I think we're all good with the error codes.